### DISCUSSION TITLE
Handling files with multiple sheets

### DISCUSSION THREAD

**Discussion by ogulcan-k** [marianfoo/ui5-cc-excelUpload#343] [Ideas]
Hello,

It seems that the control currently only considers the first sheet from the spreadsheet file. This is understandable, as figuring out how to map entities or tables from multiple sheets could induce a fatal migraine.

However, I believe implementing this feature for standalone-mode could be feasible. I have a use-case where I need to access data from multiple sheets, so I've been experimenting and have something working on my local copy.

Here's what I've done:

[GitHub Comparison Link](https://github.com/marianfoo/ui5-cc-spreadsheetimporter/compare/main...ogulcan-eu:ui5-cc-spreadsheetimporter:main)

In essence, I added a property to the ValueData interface to store the sheet name, allowing me to retrieve it later in my application. I've combined all the columns into a single array, albeit in a rather hasty manner.

I won't be submitting a pull request for this, as I'm sure there's a more elegant way to implement this feature. Additionally, I haven't run any unit tests, and my TypeScript knowledge is limited, so there may be mistakes I'm not aware of.

What are your thoughts? Do you think this feature is inline with the scope of the project?

**Comment by marianfoo** [comment]
Hi,
this ain't a bad idea and as you showed, should not be hard to implement.
I need to think about side effects this could have but should basically work as you coded it.
Let me check and if this works you can create a PR where I can continue work so you can get the contribution.
Thanks!

**Comment by marianfoo** [comment]
@ogulcan-eu  what do you think about this solution?

https://github.com/marianfoo/ui5-cc-spreadsheetimporter/pull/346

**Reply by ogulcan-k** [reply]
Looks good!

**Reply by marianfoo** [reply]
just published the new version
https://www.npmjs.com/package/ui5-cc-spreadsheetimporter/v/0.25.0

have a try and tell me if that works for you

**Reply by ogulcan-k** [reply]
Works great!

I had issues with `readAllSheets: true` not registering in the constructor, but I see you fixed that in 0.25.2

Updated and all good now
